modules/formulize/libraries/formulize-admin.js

```diff
@@ -15,19 +15,28 @@ jQuery(document).ready(function() {
         }
         hidden_section.toggle();
     });
-    
+
 });
 
-function fz_check_php_code(custom_code, block_name, icms_url, icms_userid) {
+async function fz_check_php_code(custom_code, block_name) {
+	let validateCode = new Promise(function(resolve, reject) {
     jQuery.ajax({
         type: "POST",
-        url: icms_url+"/modules/formulize/formulize_xhr_responder.php?uid="+icms_userid+"&op=validate_php_code",
+        url: window.icms_url+"/modules/formulize/formulize_xhr_responder.php?uid="+window.icms_userid+"&op=validate_php_code",
         data: {the_code: custom_code},
         success: function(result) {
             if (result.length > 0) {
-                alert("The "+block_name+" has an error:\n\n"+result+".");
-            }
-        },
-        async: false
+              resolve("The "+block_name+" has an error:\n\n"+result+".");
+            } else {
+							resolve('');
+						}
```

Comment:
The purpose of this function is to perform a validation on the given inputs and return the results. IMHO this method should not perform UX tasks like calling alert or responding with strings that are destined for end user consumption. That should be the responsibility of the calling method.

Additionally I believe the response data from this method should be an object and not a string. 

```
resolve({
  valid: result.length > 0 : false : true
  result
})
```

Reply:
This is addressed in the latest commit. Thanks!

Reply:
@arvinsingla Take a peek pls